components/lifesensor_macro/include/macro_task.h

```diff
@@ -0,0 +1,72 @@
+#ifndef __MACRO_TASK__
+#define __MACRO_TASK__
+
+#include <freertos/task.h>
+
+#define MACRO_DEFAULT_STACK_SIZE 0x1000
+/**
+ * struct lifesensor_task_t - generic task struct
+ * @tcb:    Holds the FreeRTOS tcb.
+ * @handle: Provides a reference to the task.
+ * @stack:  Stores the address of the tasks' stack.
+ *
+ * This struct serves as an abstraction for operating with FreeRTOS tasks such
+ * that it includes necessary fields for managing a tasks data. It is intended
+ * to be used together with the LIFESENSOR_TASK() macro.
+ */
+typedef struct {
+	StaticTask_t tcb;
+	TaskHandle_t handle;
+	StackType_t stack[];
```

Comment:
```suggestion
	TaskHandle_t handle;
	StaticTask_t tcb;
	StackType_t stack[];
```
would move up the handle so the whole struct can be casted directly to the handle